pkg/kubelet/dockertools/manager.go

```diff
@@ -1617,7 +1618,21 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, runningPod kub
 				// RestartPolicy::Always, but it's not a big deal.
 				glog.V(3).Infof("Container %+v is dead, but RestartPolicy says that we should restart it.", container)
 				containersToStart[index] = empty{}
+
+			}
+
+			// Trying to get the reference for container,
+			// if failing to get the ref giv the Pod as reference
+			var containerOrPodRef runtime.Object
+			containerOrPodRef = pod // default
+			ref, err := kubecontainer.GenerateContainerRef(pod, &container)
+			if ref != nil && err == nil {
+				containerOrPodRef = ref
 			}
+
+			// sending the event "died" weither or not the container should be restarted
+			dm.recorder.Eventf(containerOrPodRef, "Died", "Container \"%v\" in pod \"%v\" died.", container.Name, podFullName)
```

Comment:
@ejemba, I don't have a good solution in mind that wouldn't need a bit of overhaul of the code. I don't think we want to do that for v1.1.

We could extend the event generator in #13390 to send container events. This would have to wait until my PRs went in though.